Fixed migrate defect, allow migrate the empty directory.

andy-zhangtao opened the pull request:
If there is an empty directory exists in the $DOCKER_ROOT/containers ,
that will cause migrate fail. Then docker daemon will initialize failed.
This commit allow skips the empty directory.

Commit:
Fixed migrate defect, allow migrate the empty directory.

If there is an empty directory exists in the $DOCKER_ROOT/containers ,
that will cause migrate fail. Then docker daemon will initialize failed.
This commit allow skips the empty directory.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:andy-zhangtao/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

Commit:
Revert "Fixed migrate defect, allow migrate the empty directory."

This reverts commit a3a5a4a71a16206bff626ba9bc8c69fe7700bee2.

Commit:
Fixed migrate defect, allow migrate the empty directory

If there is an empty directory exists in the $DOCKER_ROOT/containers ,
that will cause migrate fail. Then docker daemon will initialize failed.
This commit allow skips the empty directory.
Signed-off-by: Andy Zhang <andy.zhangtao@hotmail.com>

tonistiigi commented:
Please force-push the branch without the reverted commit.

aaronlehmann commented on migrate/v1/migratev1.go:
```diff
@@ -238,12 +238,14 @@ func migrateContainers(root string, ls graphIDMounter, is image.Store, imageMapp
 		containerJSON, err := ioutil.ReadFile(filepath.Join(containersDir, id, configFileNameLegacy))
 		if err != nil {
-			return err
+			logrus.Errorf("Migrate Container Error: %v",err)
```
These error messages should be lowercase.

icecrime commented:
Closing in favor of #19172.
